### DISCUSSION TITLE
Handling two clusters with same path in YARP - how to route API requests correctly?

### DISCUSSION THREAD

**Discussion by maneshachanassery** [dotnet/yarp#2610] [Q&A] [answered]
Hi,
 
I'm working with YARP and have a setup where two clusters have the same route path, but they need to forward to different service.
Am getting multiple matches error when testing these through postman. How to create an api request call to get data from each cluster route path ? Thanks.
 
appsettings.json
 
"ReverseProxy": {
  "Routes": {
    "req1_route1": {
      "ClusterId": "clusterId1",
      "Match": {
        "Path": "/path1/{**remainder}",
        "Methods": [ "GET", "POST" ]
      }
    },
    "req2_route1": {
      "ClusterId": "clusterId2",
      "Match": {
        "Path": "/path1/{**remainder}",
        "Methods": [ "GET", "POST" ]
      }
    }
   },
   "Clusters": {
  "clusterId1": {
    "Destinations": {
      "destination1": {
        "Address": "[https://exampleone.com:1001/Name"](https://exampleone.com:1001/Name%22)
      }
    }
  },
  "clusterId2": {
    "Destinations": {
      "destination1": {
        "Address": "[https://exampleone.com:1002/Name"](https://exampleone.com:1002/Name%22)
      }
    }
  }
}
}

**Comment by Tratcher** [comment]
>  but they need to forward to different service

What criteria should be used to distinguish between the routes? Host name, query, headers, random, etc.?

**Reply by maneshachanassery** [reply]
What will be the best option to use to distinguish between the routes if the port number is the only difference between two clusters ?

**Reply by Tratcher** [reply]
For this setup routing needs there to be a difference between the requests, not the clusters.

Why are you setting it up this way? You could instead put both of those destinations in once cluster.

**Reply by maneshachanassery** [reply]
Clusters set up for two different servers, some service are same but the data in each one is different . So If I use host name to differentiate between the routes, both destinations can be added in single cluster.

**Reply by MihaZupan** [reply, marked as answer]
I'm not sure I completely understand your setup, but if you want to route requests differently depending on their host header, you can specify the `Hosts` just as you're already doing with `Path` and `Methods`, e.g.
```diff
"req1_route1": {
  "ClusterId": "clusterId1",
  "Match": {
    "Path": "/path1/{**remainder}",
    "Methods": [ "GET", "POST" ],
+   "Hosts": [ "www.foo" ]
  }
},
"req2_route1": {
  "ClusterId": "clusterId2",
  "Match": {
    "Path": "/path1/{**remainder}",
    "Methods": [ "GET", "POST" ],
+   "Hosts": [ "www.bar" ]
  }
},
```
